x-pack/plugins/task_manager/server/metrics/task_run_metrics_aggregator.ts

```diff
@@ -91,16 +90,29 @@ export class TaskRunMetricsAggregator implements ITaskMetricsAggregator<TaskRunM
   }
 
   private processTaskRunEvent(taskEvent: TaskRun) {
-    const { task, isExpired }: RanTask | ErroredTask = unwrap(taskEvent.event);
+    const taskRunResult: RanTask | ErroredTask = unwrap(taskEvent.event);
+    const { task, isExpired } = taskRunResult;
     const success = isOk((taskEvent as TaskRun).event);
     const taskType = task.taskType.replaceAll('.', '__');
     const taskTypeGroup = getTaskTypeGroup(taskType);
+    let ignoreError = false;
 
     // increment the total counters
     this.incrementCounters(TaskRunKeys.TOTAL, taskType, taskTypeGroup);
 
+    if (!success) {
+      if (isUserError((taskRunResult as ErroredTask).error)) {
+        ignoreError = true;
+        // increment the user error counters
+        this.incrementCounters(TaskRunKeys.USER_ERRORS, taskType, taskTypeGroup);
+      } else {
+        // increment the framework error counters
+        this.incrementCounters(TaskRunKeys.FRAMEWORK_ERRORS, taskType, taskTypeGroup);
+      }
+    }
+
     // increment success counters
-    if (success) {
+    if (success || ignoreError) {
       this.incrementCounters(TaskRunKeys.SUCCESS, taskType, taskTypeGroup);
```

Comment:
Here we add user_errors to `success` metric in order to subtract user_erros from SLO target.
Rather than adding them here, we can use `user_errors` metric in SLO (success + user_errors) as well.

Please let me know WDYT. 
I can remove `ignoreError` if we decide to do the addition in the SLO.

Reply:
I prefer the `use user_errors metric in SLO (success + user_errors)` option because we are trying to categorize the `errors` regardless of the type. This will allow various SLOs to determine whether or not a specific type of error is considered success for them or not.

For example:
- Alerting framework success in the future would exclude user errors and rule type implementation errors
- Rule type errors would exclude framework and user errors

This allows a breakdown down the line as mentioned here: https://github.com/elastic/kibana/issues/167055